sdk/src/transaction_context.rs

```diff
@@ -78,6 +83,7 @@ impl TransactionContext {
             return_data: TransactionReturnData::default(),
             accounts_resize_delta: RefCell::new(0),
             rent,
+            is_cap_accounts_data_size_per_transaction_enabled,
```

Comment:
> Since a new parameter was added to TransactionContext::new(), there are a lot of files touched in this PR. Those changes are hopefully simple review. If it would be helpful, I can move those changes into a separate commit within this PR.

An alternative to "permanently" adding a new parameter to `TransactionContext::new()` would be to add a method `TransactionContext::enable_cap_accounts_data_size_per_transaction()` which would be called in the bank after `TransactionContext::new()` and in `mock_process_instruction()` as well as `with_mock_invoke_context()`. That way the cleanup after the feature gate is through would be a lot easier too.